src/main/java/seedu/address/model/tag/TagList.java

```diff
@@ -0,0 +1,138 @@
+package seedu.address.model.tag;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+import seedu.address.model.person.exceptions.PersonNotFoundException;
+
+
+// To implement:
+// Class: Tag, Name, Email, Gender, DuplicateTagException, TagNotFoundException
+// Method: isSameTagType, isSameTag
+public class TagList implements Iterable<Tag> {
+
+    private final ObservableList<Tag> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Tag> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent tag as the given argument.
+     */
+    public boolean contains(Tag toCheck) {
+        requireNonNull(toCheck);
+        // a tutor should only have 1 name, email and gender.
+        if (toCheck instanceof Name || toCheck instanceof Email || toCheck instanceof Gender) {
```

Comment:
This line may not be required, as referenced to #12, `Tag` is not a superclass of `Name`, `Email` and `Gender`.

Reply:
Thank you. This part is deleted.